k8s/rke2-mc/masters.tf

```diff
@@ -0,0 +1,113 @@
+resource "aws_lb_target_group" "kube_master" {
+  name     = "${var.environment}-kube-masters"
+  port     = var.k8s_api_server_port
+  protocol = "TCP"
+  vpc_id   = var.primary_vpc_id
+
+  health_check {
+    protocol            = "TCP"
+    healthy_threshold   = 5
+    unhealthy_threshold = 5
+  }
+}
+
+resource "aws_lb_listener" "kube_master" {
+  default_action {
+    target_group_arn = aws_lb_target_group.kube_master.arn
+    type             = "forward"
+  }
+
+  load_balancer_arn = var.master_load_balancer_id
+  port              = var.k8s_api_server_port
+  protocol          = "TCP"
+}
+
+data "aws_instance" "seeder" {
+  filter {
+    name   = "tag:Role"
+    values = ["seeder"]
+  }
+
+  depends_on = [module.seeder_asg]
+}
+
+locals {
+  master_lb_hostname = var.master_load_balancer_internal_dns != "" ? split(".", var.master_load_balancer_internal_dns)[0] : ""
+  seeder_url         = "https://${data.aws_instance.seeder.private_ip}:9345"
+
+  rke_san = [
+    var.master_load_balancer_public_ip,
+    var.master_load_balancer_private_ip,
+    local.master_lb_hostname,
+    var.master_load_balancer_internal_dns
+  ]
+}
+
+resource "random_uuid" "random_cluster_id" {}
+
+module "seeder_asg" {
```

Comment:
that could be an issue.
if the instance will be terminated, a new instance will create and a new RKE cluster with it. no?